agent/consul/leader_metrics.go

```diff
@@ -130,24 +134,35 @@ func (m CertExpirationMonitor) Monitor(ctx context.Context) error {
 
 	logger := m.Logger.With("metric", strings.Join(m.Key, "."))
 
+	fn := func() {
```

Comment:
can't think of a reason not to use a function literal here but a small nit was to see if we could rename the literal to something like `emitCertExpiryFunc` or some other descriptive name. Wdyt?